src/awkward/operations/structure.py

```diff
@@ -2226,9 +2226,10 @@ def apply(layout, depth, posaxis):
             ),
         ):
             new_layout = layout.toListOffsetArray64(True)
+            new_length = new_layout.stops[-1]
```

Comment:
Oh, I just noticed another thing: if the length of this ListOffsetArray is zero, the `stops` will be, too.  The `offsets` length would always be at least one, though, so this one would be okay with `stops` replaced by `offsets`.

Reply:
(An empty ListOffsetArray's `offsets` has length 1, though the value of this `offsets[0]` is only guaranteed to be zero because the previous line ensured that the `offsets` starts with 0—the `True` in `toListOffsetArray`.)

Reply:
I'm at my computer now (no longer emailing on my phone), so I'll get this one.

Reply:
Done.

Reply:
> Oh, I just noticed another thing: if the length of this ListOffsetArray is zero, the `stops` will be, too. The `offsets` length would always be at least one, though, so this one would be okay with `stops` replaced by `offsets`.

Ah I didn't know that! Thanks for the catch. I'm a bit conscious of making these PRs when I'm not fully up to speed with every detail so i appreciate the input